mitiq/zne/inference.py

```diff
@@ -1637,6 +1639,145 @@ def reduce(self) -> float:
+            scale factor.
+        """
+
+        return a + b * np.exp(-1.0 * c * scale_factor)
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+
+        float,
+    ]:
+        """Static method which evaluates an exponential extrapolation to the
+        zero-noise limit using Bayesian inference.
+
+        Args:
+            scale_factors: The array of noise scale factors.
+            exp_values: The array of expectation values.
+            full_output: If False (default), only the zero-noise limit is
+                returned. If True, additional information about the
+                extrapolated limit is returned too.
+        Returns:
+            zne_limit: The extrapolated zero-noise limit. If "full_output"
+                is False (default value), only this parameter is returned.
+            opt_params: The parameter array of the best fitting model.
+            zne_curve: The callable function which best fit the input data.
+                It maps a real noise scale factor to a real expectation value.
+                It is equal "zne_limit" when evaluated at zero.
+
+        Note:
+            This method computes the zero-noise limit only from the information
+            contained in the input arguments. To extrapolate from the internal
+            data of an instantiated Factory object, the bound method
+            ".reduce()" should be called instead.
+        """
+        with pm.Model():
+            """
+            We assumme that the priors for the model parameters is a uniform
+            distribution with an upper limit 1 and a lower limit 0,
+            except for b with a lower limit -1.
+            """
+            a = pm.Uniform('a', 0, 1)
+            b = pm.Uniform('b', -1, 1)
+            c = pm.Uniform('c', 0, 1)
+            eps = pm.Uniform('eps', 0, 0.5)
+
+            pm.Normal(
+                'expval',
+                mu=ExpBayesFactory._exp_ansatz(a, b, c, scale_factors),
+                sd=eps,
+                observed=exp_values,
+            )
+
+            trace = pm.sample(target_accept=0.95)
+
+        # Optimal model parameters:
+        a = trace['a'].mean()
+        b = trace['b'].mean()
+        c = trace['c'].mean()
+        opt_params, zne_error = [a, b, c], trace['eps'].mean()
+
+        def zne_curve(scale_factor: float) -> float:
+            return ExpBayesFactory._exp_ansatz(a, b, c, scale_factor)
+
+        zne_limit = ExpBayesFactory._exp_ansatz(a, b, c, 0.0)
+        if not full_output:
+            return zne_limit
+
+        params_cov = None
+        return zne_limit, zne_error, opt_params, params_cov, zne_curve
+
+    def reduce(self) -> float:
+        """Evaluates the zero-noise limit found by fitting a polynomial of degree
+        `self.order` to the internal data stored in the factory.
```

Comment:
Sorry I missed this before: this docstring of `ExpBayesFactory.reduce` speaks about a poly fit but this is wrong.
We know that when #470 will be merged this method will disappear anyway. So, for the moment, you could simply remove the docstring just to avoid confusion.

Reply:
Indeed, removed.